[Bug] [Transform-V2] Milvus create collection error because of vector field dimension is null

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.


### What happened

I added a transform that is embedding value as a vector field then write to milvus, but vector field dimension is null causes create collection error

### SeaTunnel Version

dev/2.3.12-SNAPSHOT

### SeaTunnel Config

```conf
env {
  parallelism = 1
  job.mode = "BATCH"
}

source {
  S3File {
    path = "/seatunnel/test_csv_data.csv"
    bucket = "s3a://lthen"
    fs.s3a.endpoint="tos-s3-cn-beijing.volces.com"
    fs.s3a.aws.credentials.provider="org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider"
    file_format_type = "csv"
    access_key="xxx"
    secret_key="xxx",
    csv_use_header_line = true,
    field_delimiter = ","
    schema={
        fields {
            id = int
            code = int
            data = string
            success = boolean
        },
        primaryKey {
            name = "id"
            columnNames = ["id"]
        }
    }
  }
}

transform {
  Embedding {
    model_provider = "DOUBAO"
    model = "doubao-embedding-text-240715"
    api_key = "xxx"
    secret_key = "xxx"
    vectorization_fields {
        data_vector = data
    },
    custom_config={
      custom_response_parse = "$.data[*].embedding"
      custom_request_headers = {
          "Content-Type"= "application/json"
          "Authorization"= "Bearer ${api_key}"
      }
      custom_request_body ={
          model = "${model}"
          input = ["${input}"]
      }
    }
  }
}

sink {
  Milvus {
    url = "http://127.0.0.1:19530"
    token = "root:xxx"
    batch_size = 1000
    enable_auto_id=true,
    schema_save_mode=RECREATE_SCHEMA
  }
}
```

### Running Command

```shell
SeaTunnelEngineLocalExample#main
```

### Error Exception

```log
Caused by: org.apache.seatunnel.engine.common.exception.JobException: org.apache.seatunnel.common.exception.SeaTunnelRuntimeException: ErrorCode:[API-09], ErrorDescription:[Handle save mode failed]
	at org.apache.seatunnel.engine.server.master.JobMaster.handleSaveMode(JobMaster.java:573)
	at org.apache.seatunnel.engine.server.master.JobMaster.lambda$init$1(JobMaster.java:273)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.apache.seatunnel.engine.server.master.JobMaster.init(JobMaster.java:267)
	at org.apache.seatunnel.engine.server.CoordinatorService.lambda$submitJob$6(CoordinatorService.java:656)
	at org.apache.seatunnel.api.tracing.MDCRunnable.run(MDCRunnable.java:43)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.apache.seatunnel.connectors.seatunnel.milvus.exception.MilvusConnectorException: ErrorCode:[MILVUS-14], ErrorDescription:[Create collection error]
	at org.apache.seatunnel.connectors.seatunnel.milvus.catalog.MilvusCatalog.createTableInternal(MilvusCatalog.java:308)
	at org.apache.seatunnel.connectors.seatunnel.milvus.catalog.MilvusCatalog.createTable(MilvusCatalog.java:200)
	at org.apache.seatunnel.api.sink.DefaultSaveModeHandler.createTable(DefaultSaveModeHandler.java:213)
	at org.apache.seatunnel.api.sink.DefaultSaveModeHandler.recreateSchema(DefaultSaveModeHandler.java:134)
	at org.apache.seatunnel.api.sink.DefaultSaveModeHandler.handleSchemaSaveMode(DefaultSaveModeHandler.java:85)
	at org.apache.seatunnel.api.sink.SaveModeHandler.handleSaveMode(SaveModeHandler.java:42)
	at org.apache.seatunnel.api.sink.SaveModeExecuteWrapper.execute(SaveModeExecuteWrapper.java:36)
	at org.apache.seatunnel.engine.server.master.JobMaster.handleSaveMode(JobMaster.java:568)
	... 20 more
Caused by: java.lang.NullPointerException: dimension is marked non-null but is null
	at io.milvus.param.collection.FieldType$Builder.withDimension(FieldType.java:201)
	at org.apache.seatunnel.connectors.seatunnel.milvus.utils.sink.MilvusSinkConverter.convertToFieldType(MilvusSinkConverter.java:181)
	at org.apache.seatunnel.connectors.seatunnel.milvus.catalog.MilvusCatalog.createTableInternal(MilvusCatalog.java:260)
	... 27 more
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/embedding/EmbeddingTransform.java
+++ b/seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/embedding/EmbeddingTransform.java
@@ -52,7 +52,7 @@ public class EmbeddingTransform extends MultipleFieldOutputTransform {
     private final ReadonlyConfig config;
     private List<String> fieldNames;
     private List<Integer> fieldOriginalIndexes;
-    private Model model;
+    private transient Model model;
     private Integer dimension;
 
     public EmbeddingTransform(
@@ -213,6 +213,7 @@ public class EmbeddingTransform extends MultipleFieldOutputTransform {
 
     @Override
     protected Column[] getOutputColumns() {
+        tryOpen();
         Column[] columns = new Column[fieldNames.size()];
         for (int i = 0; i < fieldNames.size(); i++) {
             columns[i] =

--- a/seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/embedding/EmbeddingTransform.java
+++ b/seatunnel-transforms-v2/src/main/java/org/apache/seatunnel/transform/nlpmodel/embedding/EmbeddingTransform.java
@@ -17,6 +17,8 @@
 
 package org.apache.seatunnel.transform.nlpmodel.embedding;
 
+import org.apache.seatunnel.shade.com.google.common.annotations.VisibleForTesting;
+
 import org.apache.seatunnel.api.configuration.ReadonlyConfig;
 import org.apache.seatunnel.api.table.catalog.CatalogTable;
 import org.apache.seatunnel.api.table.catalog.Column;
@@ -212,7 +214,8 @@ public class EmbeddingTransform extends MultipleFieldOutputTransform {
     }
 
     @Override
-    protected Column[] getOutputColumns() {
+    @VisibleForTesting
+    public Column[] getOutputColumns() {
         tryOpen();
         Column[] columns = new Column[fieldNames.size()];
         for (int i = 0; i < fieldNames.size(); i++) {

--- a/seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/embedding/EmbeddingTransformTest.java
+++ b/seatunnel-transforms-v2/src/test/java/org/apache/seatunnel/transform/embedding/EmbeddingTransformTest.java
@@ -1,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.transform.embedding;
+
+import org.apache.seatunnel.shade.com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.seatunnel.shade.com.fasterxml.jackson.core.type.TypeReference;
+import org.apache.seatunnel.shade.com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.CatalogTableUtil;
+import org.apache.seatunnel.api.table.catalog.Column;
+import org.apache.seatunnel.transform.nlpmodel.embedding.EmbeddingTransform;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+public class EmbeddingTransformTest {
+
+    @Test
+    void testOutputColumns() throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        String sourceConfig =
+                "{\"path\":\"/seatunnel/test_csv_data.csv\",\"bucket\":\"s3a://ltchen\",\"fs.s3a.endpoint\":\"tos-s3-cn-beijing.volces.com\",\"fs.s3a.aws.credentials.provider\":\"org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider\",\"file_format_type\":\"csv\",\"access_key\":\"xxx\",\"secret_key\":\"xxx\",\"csv_use_header_line\":true,\"field_delimiter\":\",\",\"schema\":{\"fields\":{\"id\":\"int\",\"code\":\"int\",\"data\":\"string\",\"success\":\"boolean\"},\"primaryKey\":{\"name\":\"id\",\"columnNames\":[\"id\"]}},\"plugin_name\":\"S3File\"}";
+        Map<String, Object> sourceConfigMap =
+                objectMapper.readValue(sourceConfig, new TypeReference<Map<String, Object>>() {});
+        ReadonlyConfig readonlyConfig = ReadonlyConfig.fromMap(sourceConfigMap);
+        CatalogTable inputCatalogTable = CatalogTableUtil.buildWithConfig("S3File", readonlyConfig);
+
+        int dimension = 1024;
+        String embeddingConfig =
+                "{\"model_provider\":\"AMAZON\",\"model\":\"amazon.titan-embed-text-v2:0\",\"aws_region\": \"us-east-1\", \"api_key\":\"xxx\",\"secret_key\":\"xxx\",\"api_path\": \"https://aws.amazon.com/bedrock/amazon-models\", \"dimension\": "
+                        + dimension
+                        + ",\"vectorization_fields\":{\"data_vector\":\"data\"},\"plugin_name\":\"Embedding\"}";
+        Map<String, Object> embeddingConfigMap =
+                objectMapper.readValue(
+                        embeddingConfig, new TypeReference<Map<String, Object>>() {});
+        ReadonlyConfig config = ReadonlyConfig.fromMap(embeddingConfigMap);
+        EmbeddingTransform embeddingTransform = new EmbeddingTransform(config, inputCatalogTable);
+
+        Column[] columns = embeddingTransform.getOutputColumns();
+        for (Column column : columns) {
+            Assertions.assertEquals(dimension, column.getScale());
+        }
+    }
+}

